packages/components/src/search-control/index.native.js

```diff
@@ -1,96 +1,237 @@
 /**
  * External dependencies
  */
-import { TextInput, View, TouchableHighlight } from 'react-native';
+import {
+	TextInput,
+	Text,
+	View,
+	TouchableOpacity,
+	Platform,
+	useColorScheme,
+} from 'react-native';
 
 /**
  * WordPress dependencies
  */
-import { useState, useRef } from '@wordpress/element';
-import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { useState, useRef, useMemo, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { ToolbarButton } from '@wordpress/components';
+import { Button, Gridicons } from '@wordpress/components';
 import {
 	Icon,
-	cancelCircleFilled,
-	arrowLeft,
-	search as searchIcon,
+	cancelCircleFilled as cancelCircleFilledIcon,
+	arrowLeft as arrowLeftIcon,
+	close as closeIcon,
 } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import styles from './style.scss';
+import allStyles from './style.scss';
+import platformStyles from './platform-style.scss';
+
+// Merge platform specific styles with the default styles.
+const baseStyles = { ...allStyles };
+for ( const selector in platformStyles ) {
+	baseStyles[ selector ] = {
+		...baseStyles[ selector ],
+		...platformStyles[ selector ],
+	};
+}
+
+function selectModifiedStyles( styles, modifier ) {
+	const modifierMatcher = new RegExp( `--${ modifier }$` );
+	const modifierSelectors = Object.keys( styles ).filter( ( selector ) =>
+		selector.match( modifierMatcher )
+	);
+
+	return modifierSelectors.reduce( ( modifiedStyles, modifierSelector ) => {
+		const blockElementSelector = modifierSelector.split( '--' )[ 0 ];
+		modifiedStyles[ blockElementSelector ] = styles[ modifierSelector ];
+		return modifiedStyles;
+	}, {} );
+}
+
+function mergeStyles( styles, updateStyles, selectors ) {
+	selectors.forEach( ( selector ) => {
+		styles[ selector ] = {
+			...styles[ selector ],
+			...updateStyles[ selector ],
+		};
+	} );
+
+	return styles;
+}
 
 function SearchControl( {
 	value,
 	onChange,
-	label,
-	placeholder = __( 'Search' ),
+	placeholder = __( 'Search Blocks' ),
 } ) {
 	const [ isActive, setIsActive ] = useState( false );
+	const [ currentStyles, setCurrentStyles ] = useState( baseStyles );
 
+	const isDark = useColorScheme() === 'dark';
 	const inputRef = useRef();
 
-	const searchFormStyle = usePreferredColorSchemeStyle(
-		styles.searchForm,
-		styles.searchFormDark
-	);
+	const isIOS = Platform.OS === 'ios';
 
-	const searchFormInputStyle = usePreferredColorSchemeStyle(
-		styles.searchFormInput,
-		styles.searchFormInputDark
-	);
+	const darkStyles = useMemo( () => {
+		return selectModifiedStyles( baseStyles, 'dark' );
+	}, [] );
```

Comment:
If this is a static object that won't change, I'm wondering if it would make sense to have it as constant outside the functional component, as we have the base styles, wdyt?

The same comments apply to the rest of the styles:
- darkStyles
- activeStyles
- activeDarkStyles

Reply:
I noticed that but left these here to show the evolution on extracting the pattern to a hook. Here is the abstraction around how the modified styles are selected from the base styles: https://github.com/WordPress/gutenberg/commit/5004d705ace202ce55c4ca652ae7d21085672cd1#diff-03ee4c0d07daa51fb0c8182f80cdd50c7f5317727593a6a75ea3f877c744f610R96 

I don't mind moving these outside of the component but figure `useMemo` should give us enough cacheing to be performant.

Reply:
Ok I see, then we can leave them here 👍, thanks for the clarification 🙇‍♂️.